lib/charms/mongodb/v0/mongodb_backups.py

```diff
@@ -64,8 +69,43 @@ def _on_pbm_config_changed(self, event) -> None:
             event.defer()
             return
 
-        # TODO handle PBM configurations set by the user
-        self.charm.unit.status = ActiveStatus()
+        # URI is set with `snap set`
+        pbm_snap.set({"uri": self._backup_config.uri})
+
+        # presets for PBM snap configurations
+        pbm_configs = {}
+        pbm_configs["storage.type"] = "s3"
+        pbm_configs["storage.s3.serverSideEncryption.sseAlgorithm"] = "aws:kms"
+
+        credentials = self.s3_client.get_s3_connection_info()
+        for s3_option, s3_value in credentials.items():
+            if s3_option not in S3_PBM_OPTION_MAP:
+                continue
+
+            pbm_name = S3_PBM_OPTION_MAP[s3_option]
+            pbm_configs[pbm_name] = s3_value
+
+        for (pbm_key, pbm_value) in pbm_configs.items():
+            try:
+                self._pbm_set_config(pbm_key, pbm_value)
+            except subprocess.CalledProcessError as e:
+                logger.error(
+                    "Failed to configure the PBM snap with key=value %s=%s, failed with error: %s",
+                    str(pbm_key),
+                    str(pbm_value),
```

Comment:
we shouldn't print credentials to the logs 😞

Reply:
🤦‍♀️ second time making this mistake, thanks for catching Mykola